### PR TITLE
feat: Add the ability to save filters for contact

### LINKED ISSUE #3790: Add the ability to save custom filters as segments.

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

Fixes #3790 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/contacts/components/ContactsView.vue [hunk 1]
```diff
@@ -93,6 +102,9 @@ export default {
           `CONTACTS_FILTER.ATTRIBUTES.${filter.attributeI18nKey}`
         ),
       })),
+      customViewsQuery: {},
+      filterType: 1,
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/contacts/components/ContactsView.vue:106]: Use this as a constant with name of constant referring to type of filter.

File: app/javascript/dashboard/routes/dashboard/contacts/components/ContactsView.vue [hunk 2]
```diff
@@ -218,6 +236,7 @@ export default {
     },
     onApplyFilter(payload) {
       this.closeContactInfoPanel();
+      this.customViewsQuery = { payload: payload };
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/contacts/components/ContactsView.vue:239]:
```suggestion
      this.customViewsQuery = { payload };
```

File: app/javascript/dashboard/routes/dashboard/contacts/components/Header.vue [hunk 3]
```diff
@@ -38,6 +38,15 @@
             {{ $t('CONTACTS_PAGE.FILTER_CONTACTS') }}
           </woot-button>
         </div>
+        <woot-button
+          v-if="hasAppliedFilters"
+          class="margin-right-small clear"
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/contacts/components/Header.vue:43]: User variant for clear

**Review by nithindavid — CHANGES_REQUESTED** [review]

**Review by tejaswinichile — APPROVED** [review]
LGTM 👍🏼

File: app/javascript/dashboard/routes/dashboard/contacts/components/ContactsView.vue [hunk 4]
```diff
@@ -10,6 +10,7 @@
         :on-toggle-import="onToggleImport"
         :on-toggle-filter="onToggleFilters"
         :header-title="label"
+        @open-customview-modal="onClickOpenCustomViewsModal"
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/contacts/components/ContactsView.vue:13]:
```suggestion
        @on-toggle-save-filter="onToggleSaveFilters"
```

**Review by nithindavid — APPROVED** [review]

**Pull request merged**